Reflection_Engine/Create/MethodBase.cs

```diff
@@ -66,5 +66,25 @@ public static MethodBase MethodBase(Type type, string methodName, List<string> p
         }
 
         /*******************************************/
+
+        public static MethodBase MethodBase(Type type, string methodName, List<Type> paramTypes)
+        {
+            MethodBase method;
+            if (methodName == ".ctor")
+                method = type.GetConstructor(paramTypes.ToArray());
+            else
+                method = type.GetMethod(methodName, paramTypes.ToArray());
+
+            // the above will return null if the type is a generic type
+            // that is because we serialise a generic method with its constraints
+            // rather than serialising the generics
+            if (method != null)
+                return method;
+
+            // So, let's try the other overload
+            return MethodBase(type, methodName, paramTypes.Select(x => x.Name).ToList());
```

Comment:
Uses the old overload in case the new fails when dealing with generic types (I believe around 10% of the cases, maybe less)